manifest.md

```diff
@@ -60,17 +60,22 @@ A client will distinguish a manifest list from an image manifest based on the Co
 
         This OPTIONAL property specifies an array of strings, each specifying a mandatory CPU feature (for example `sse4` or `aes`).
 
-- **`annotations`** *string-string hashmap*
+- **`annotations`** *string-string map*
 
     This OPTIONAL property contains arbitrary metadata for the manifest list.
-    Annotations is a key-value, unordered hashmap.
-    Keys are unique, and best practice is to namespace the keys.
-    Common annotation keys include:
-    * **created** date on which the image was built (string, timestamps type)
-    * **authors** contact details of the people or organization responsible for the image (freeform string)
-    * **homepage** URL to find more information on the image (string, must be a URL with scheme HTTP or HTTPS)
-    * **documentation** URL to get documentation on the image (string, must be a URL with scheme HTTP or HTTPS)
+    Annotations MUST be a key-value map where both the key and value MUST be strings.
+    While the value MUST be present, it MAY be an empty string.
+    Keys MUST be unique within this map, and best practice is to namespace the keys.
+    Keys SHOULD be named using a reverse domain notation - e.g. `com.example.myKey`.
+    Keys using the `org.opencontainers` namespace are reserved and MUST NOT be used by other specifications.
+    If there are no annotations then this property MUST either be absent or be an empty map.
```

Comment:
On Wed, Sep 07, 2016 at 11:10:55AM -0700, Doug Davis wrote:
> > +    If there are no annotations then this property MUST either be absent or be an empty map.
>
> … What I was trying to avoid was someone claiming an impl is not
> compliant because they put an empty map instead of dropping the
> property all together.

Under what grounds would they make that claim?  There is no “MUST
contain at least one key” requirement.

> OPTIONAL only covers the ability to leave it off entirely, not for
> it to be empty.

Agreed.  The empty map case is covered by there not being any required
keys [1], because the spec'ed type is a JSON object, and an empty
object is still an object.  Array/object/string properties allow the
empty value by default unless the spec explicitly forbids it.

[1]: https://github.com/opencontainers/image-spec/pull/164/files/7350d6ca92e444636f45b5c669076cb83dee6756#r77871484